flow/core/kernel/vehicle/traci.py

```diff
@@ -949,18 +950,22 @@ def _prev_edge_followers(self, veh_id, edge_dict, lane, num_edges):
 
         return tailway, follower
 
-    def apply_acceleration(self, veh_ids, acc):
+    def apply_acceleration(self, veh_ids, acc, smooth=False):
         """See parent class."""
-        # to hand the case of a single vehicle
+        # to handle the case of a single vehicle
         if type(veh_ids) == str:
             veh_ids = [veh_ids]
             acc = [acc]
 
         for i, vid in enumerate(veh_ids):
             if acc[i] is not None and vid in self.get_ids():
```

Comment:
Btw this `in` operates in O(n), maybe it would lead to a little speedup to change `self.__ids` from a list to a set so that it operates in O(1)

Reply:
@eugenevinitsky I'll leave this to you to decide.

Reply:
This is probably true for many things. We can look into it if it makes things faster (post libsumo). Right now it's probably over-optimization.